metals/src/main/scala/scala/meta/internal/metals/WorkspaceLspService.scala

```diff
@@ -1316,6 +1315,10 @@ class Folder(
   lazy val isMetalsProject: Boolean =
     isKnownMetalsProject || path.resolve(".metals").exists || path
       .isMetalsProject()
+
+  lazy val optDelegatePath: Option[AbsolutePath] =
```

Comment:
Do we need to save it actually, is it problematic if we always search buildTargets if no service corresponds to a path?

Reply:
Searching in `buildTargets` won't  work if it is before import, so if we want the user to be able to open both main project and project ref at the same time, then yes. (And currently this is the case for any workspace folder remove, since it reloads everything).

Reply:
makes sense, thanks for explaining. Might be worth mentioning in a comment

Reply:
Also maybe mention what is the path exactly.

Reply:
It's good, thanks!